src/BondEvaluatorQuartic.h

```diff
@@ -0,0 +1,236 @@
+            if (lj1 != Scalar(0) && r6inv > sigma6inv / Scalar(2.0))
+                {
+                Scalar epsilon = lj2 * lj2 / Scalar(4.0) / lj1;
+                force_divr = r2inv * r6inv * (Scalar(12.0) * lj1 * r6inv - Scalar(6.0) * lj2);
+                bond_eng = r6inv * (lj1 * r6inv - lj2) + epsilon;
+                }
+            else
+                {
+                force_divr = 0;
+                bond_eng = 0;
+                }
+
+            // Quartic component
+            // If the distance is less than the quartic cutoff distance, calculate as normal
+            if (rsq < r_0 * r_0)
+                {
+                Scalar r_red = fast::sqrt(rsq) - r_0;
+                force_divr += Scalar(-1) * k * r_red
+                              * (4 * r_red * r_red - 3 * (b_1 + b_2) * r_red + 2 * b_1 * b_2)
+                              / (r_red + r_0);
+                bond_eng += k * (r_red - b_1) * (r_red - b_2) * r_red * r_red + U_0;
+                }
+            else
+                {
+                force_divr += 0;
+                bond_eng += U_0;
+                }
+            }
+        else // case when delta != 0, a square root needs to be taken
+            {
+            r = fast::sqrt(rsq) - delta;
+            const Scalar r2inv = Scalar(1.0) / r / r;
+            const Scalar r6inv = r2inv * r2inv * r2inv;
+            const Scalar sigma6inv = lj2 / lj1;
+
+            // wca cutoff: r < 2^(1/6)*sigma
+            // if epsilon or sigma is zero OR r is beyond cutoff, the WCA force and energy are zero
+            if (lj1 != Scalar(0) && r6inv > sigma6inv / Scalar(2.0))
+                {
+                force_divr
+                    = r6inv * (Scalar(12.0) * lj1 * r6inv - Scalar(6.0) * lj2) / r / (r + delta);
+                bond_eng = r6inv * (lj1 * r6inv - lj2) + epsilon;
+                }
+            else
+                {
+                force_divr = 0;
+                bond_eng = 0;
+                }
+
+            // Quartic component of the force
+            if (r < r_0)
+                {
+                Scalar r_red = r - r_0;
+                force_divr += Scalar(-1) * k * r_red
+                              * (4 * r_red * r_red - 3 * (b_1 + b_2) * r_red + 2 * b_1 * b_2)
+                              / (r_red + r_0 + delta);
+                bond_eng += k * (r_red - b_1) * (r_red - b_2) * r_red * r_red + U_0;
+                }
+            else
+                {
+                force_divr += 0;
+                bond_eng += U_0;
+                }
+            }
+
+        return true;
+        }
+
+#ifndef __HIPCC__
+    static std::string getName()
+        {
+        return std::string("Quartic");
+        }
+#endif
+
+    private:
+    Scalar k;       //!< k parameter for quartic potential strength
+    Scalar r_0;     //!< energy barrier breaking distance
+    Scalar b_1;     //!< quartic tuning parameter #1
+    Scalar b_2;     //!< quartic tuning parameter #2
```

Comment:
Can these parameters (b_1 and b_2) be described in a way that makes clear which one is which?

Reply:
Since the b_1 and b_2 are algebraically equivalent, (if you were to exchange their values, you would get the same potential) I'm not sure there's much merit to distinguishing the two. I've changed the description to "primary" and "secondary" so it's a bit easier to see when scanning the document